src/analyses/base.ml

```diff
@@ -416,7 +416,12 @@ struct
       in
       let f = function
         | Addr.Addr (x, o) -> f_addr (x, o)
-        | Addr.NullPtr -> VD.bot () (* TODO: why bot? *)
+        | Addr.NullPtr ->
```

Comment:
Should we not also produce a warning here?

Reply:
We actually already produce warnings about this, but in another place about the whole address set:
https://github.com/goblint/analyzer/blob/d915302fbd008bdc13fafcf954c5f605ffed2c7d/src/analyses/base.ml#L1017-L1022